enhancements/sig-architecture/229-manifestworkreplicaset-rollback/README.md

```diff
@@ -0,0 +1,469 @@
+    1. Work controller will add a ControllerRef in the history `.ownerReferences`.
+    1. The current history should be the largest `.revision` number amongst the others. Update `.revision` to the largest if it is not. (for example, after rollback, we will need to update it to the latest.)
+2. `(NEW)` Update `.spec.updateRevision` to the newly calculated revision.
+3. Iterate each placement references (`.spec.placementRefs`)
+    1. Get all manifestworks associated with the current placement
+    1. `(NEW)` Find the progressing manifestwork hash
+        1. Use `.spec.updateRevision` if `.status.abortTime` is not null, otherwise, use `.status.currentRevision`.
+    1. Find manifestworks which has the same hash of the progressing manifestwork hash.
+        1. [Determine cluster rollout status (Failed, Success)](https://github.com/open-cluster-management-io/ocm/blob/main/pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_reconcile.go#L96)
+    1. Create Rollout handler
+    1. Calculate RolloutResult (rollout/timeout/removed candidate clusters)
+    1. `(NEW)` Evaluate abort condition,
+        1. `.spec.placementRefs[*].rolloutStrategy.failureStrategy` is `AbortAll`
+        1. and more than `maxFailures` percentage of the current group of rollout clusters reaches to `.spec.placementRefs[*].rolloutStrategy.progressive.progressDeadline`.
+    1. `(NEW)` If the current rollout clusters are in abort condition.
+        1. Set set `.status.abortedTime` to the current time.
+        1. Fetch `ControllerRevision` corresponding to `.status.currentRevision` and extract `.spec.manifestWorkTemplate`
+        1. Apply `.spec.manifestWorkTemplate` to all clusters (`ManifestWork` resource in each cluster namespace) in `rolloutResult.clusterToRollout`
+    1. Iterate each cluster of `rolloutResult.clusterToRollout`
+        1. `(NEW)` Create `ManifestWork` and add `work.open-cluster-management.io/controller-hash` label with new hash
+        2. Apply `ManifestWork` to each cluster namespace
+    1. Clean up manifestwork resources from each cluster namespace
+    1. `(NEW)` Status update
+        1. Calculate `.summary.updated` by counting the number of `Succeeded` status of rollout clusters (which have the updateRevision)
+        1. if `.summary.updated` == `.summary.desiredTotal`,
+            1. Set `.status.updateRevision` to `.status.currentRevision`
+
+### Status transition
+
+This section describes the detail status transition for each rollout condition.
+
+#### New manifestwork update transition
+
+| Transition order | PlacementVerified (Status) | ManifestworkApplied (Status) | Progressing (Status) | PlacementRollOut (Status) | status.currentRevision | status.updateRevision | status.abortedTime |
+| --- | --- | --- | --- | --- | --- | --- | --- |
+| 1 | AsExpected (True) | | NewRevisionCreated (True) | |  | 6b6dcbd695 | |
```

Comment:
could you explain the meaning of each row and column? e.g. ManifestworkApplied is a condition type, but NewRevisionCreated is a reason or another condition type?